src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -2424,7 +2424,9 @@ ProduceBoundNode:
                         ' Deal with Optional arguments
                         ' Need to handle optional arguments here, there could be conversion errors, etc.
 
-                        argument = GetArgumentForParameterDefaultValue(param, node, diagnostics, callerInfoOpt)
+                        ' reducedExtensionReceiverOpt is used to determine the default value of a CallerArgumentExpression when it refers to the first parameter of an extension method.
+                        ' Don't bother with correctly determining the correct value for this case since we're in an error case anyway.
```

Comment:
>' Don't bother with correctly determining the correct value for this case since we're in an error case anyway. [](http://example.com/codeflow?start=24&length=110)

Are we testing behavior for this case?

Reply:
> Are we testing behavior for this case?

Could we get in trouble by referring to "wrong" parameters? Perhaps with getting additional confusing diagnostics, etc.

Reply:
Yes I believe I wrote a test that hits this code path. The test doesn't produce additional diagnostics. Certainly this doesn't prove that no additional diagnostics can be produced, but I think this sort of thing may be adjusted based on feedback. Also note that some callsites of `ReportOverloadResolutionFailureForASetOfCandidates` will not even have a "best symbol", so they will have to pass "Nothing" no matter what.